src/makeCalculator.js

```diff
@@ -4,7 +4,42 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      if (typeof callback === 'function') {
+
+        callback.call(this, number);
+
+        return this;
+      }
+    },
```

Comment:
The `operate` method should check if the passed `callback` is one of the calculator's own methods. Currently, it accepts any function, which could lead to unexpected behavior if an external function is passed. Consider adding a check to ensure that the callback is one of the calculator's methods.